src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/resource/AuditResource.kt

```diff
@@ -126,9 +127,15 @@ class AuditResource(
       )
     ]
   )
-  fun insertAuditEvent(@RequestBody auditEvent: AuditEvent) {
+  fun insertAuditEvent(@RequestHeader(value = "traceparent", required = false) traceParent: String?, @RequestBody auditEvent: AuditEvent) {
+    auditEvent.operationId?.let { traceParent?.traceId() }
```

Comment:
Don't think this is doing what you want, since the result of the `let` will be ignored.
Something like
```
    val eventWithOperationId = auditEvent.copy(operationId = auditEvent.operationId ?: traceParent?.traceId())
```
will do the trick.  Note the use of a `copy` function to prevent mutating the input variable too

Reply:
Yes - also added additional tests to ensure code logic correct